runtime/kernel/capsules/src/otp.rs

```diff
@@ -0,0 +1,192 @@
+            | hek @ reg::LOCK_HEK_PROD_7 => {
+                let hek_num_words = size_of_val(&Fuses::default().cptra_ss_lock_hek_prod_0) / 4;
+                if app.reg_index >= hek_num_words as u32 {
+                    return CommandReturn::failure(ErrorCode::INVAL);
+                }
+                let offset = match self.hek_offset(hek) {
+                    Ok(offset) => offset,
+                    Err(e) => {
+                        return CommandReturn::failure(e);
+                    }
+                };
+
+                let word_offset = offset / 4 + app.reg_index as usize;
+                match self.driver.write_word(word_offset, value) {
+                    Ok(written) if written == value => CommandReturn::success(),
+                    Ok(_) => CommandReturn::failure(ErrorCode::FAIL),
+                    Err(_) => CommandReturn::failure(ErrorCode::FAIL),
+                }
+            }
+            _ => CommandReturn::failure(ErrorCode::NOSUPPORT),
+        }) {
+            Ok(ret) => ret,
+            Err(_) => CommandReturn::failure(ErrorCode::FAIL),
+        }
+    }
+
+    fn set_reg(&self, reg: u32, index: u32, processid: ProcessId) -> CommandReturn {
+        if self
+            .apps
+            .enter(processid, |app, _| {
+                app.reg_offset = reg;
+                app.reg_index = index;
+            })
+            .is_err()
+        {
+            return CommandReturn::failure(ErrorCode::FAIL);
+        }
+        CommandReturn::success()
+    }
+
+    fn valid_hek_slot(&self, slot: u32) -> bool {
+        let slots = &reg::LOCK_HEK_PROD_ALL[..self.total_heks as usize];
+        slots.contains(&slot)
+    }
+
+    fn hek_offset(&self, slot: u32) -> Result<usize, ErrorCode> {
+        if !self.valid_hek_slot(slot) {
+            return Err(ErrorCode::INVAL);
+        }
+
+        match slot {
+            reg::LOCK_HEK_PROD_0 => Ok(fuses::CPTRA_SS_LOCK_HEK_PROD_0_BYTE_OFFSET),
+            reg::LOCK_HEK_PROD_1 => Ok(fuses::CPTRA_SS_LOCK_HEK_PROD_1_BYTE_OFFSET),
+            reg::LOCK_HEK_PROD_2 => Ok(fuses::CPTRA_SS_LOCK_HEK_PROD_2_BYTE_OFFSET),
+            reg::LOCK_HEK_PROD_3 => Ok(fuses::CPTRA_SS_LOCK_HEK_PROD_3_BYTE_OFFSET),
+            reg::LOCK_HEK_PROD_4 => Ok(fuses::CPTRA_SS_LOCK_HEK_PROD_4_BYTE_OFFSET),
+            reg::LOCK_HEK_PROD_5 => Ok(fuses::CPTRA_SS_LOCK_HEK_PROD_5_BYTE_OFFSET),
+            reg::LOCK_HEK_PROD_6 => Ok(fuses::CPTRA_SS_LOCK_HEK_PROD_6_BYTE_OFFSET),
+            reg::LOCK_HEK_PROD_7 => Ok(fuses::CPTRA_SS_LOCK_HEK_PROD_7_BYTE_OFFSET),
+            _ => Err(ErrorCode::INVAL),
+        }
+    }
+}
+
+/// Provide an interface for userland.
+impl SyscallDriver for Otp {
+    fn command(&self, cmd: usize, arg1: usize, arg2: usize, processid: ProcessId) -> CommandReturn {
+        match cmd as u32 {
+            cmd::OTP_READ => self.read_reg(processid),
+            cmd::OTP_WRITE => self.write_reg(arg1 as u32, processid),
```

Comment:
Do we need to use a subscribe / upcall for writing at least? I'm not sure how long it takes to burn this many fuses